include/swift/AST/Attr.h

```diff
@@ -435,6 +447,14 @@ class DeclAttribute : public AttributeBase {
     return getOptions(DK) & UserInaccessible;
   }
 
+  static bool isAddingBreaking(DeclAttrKind DK, /*True for ABI; false for API*/bool ABI) {
+    return getOptions(DK) & (ABI ? ABIBreakingToAdd : APIBreakingToAdd);
+  }
+
+  static bool isRemovingBreaking(DeclAttrKind DK, /*True for ABI; false for API*/bool ABI) {
```

Comment:
Can you use an enum for these rather than a boolean?

Reply:
I have trouble figuring out where the enum should be placed. What do you think about breaking it into two functions `isRemovingBreakingABI` and `isRemovingBreakingAPI`?